File: ios/Sources/Omnesis/ActivitySegments/ActivitySegmentsIcon.swift
// SPDX-License-Identifier: AGPL-3.0-or-later
// Copyright (c) 2026 Adrien Conrath

import Foundation

/// Self-owned glyph for Activity Segments, pushed to the gateway on
/// source registration — mirrors `AppleHealthIcon`'s rationale: Core
/// Motion is an Apple OS feature, not a third-party service, but it has
/// no first-party icon of its own to hotlink, so this package owns a
/// generic glyph instead.
public enum ActivitySegmentsIcon {
    /// Lucide `footprints` (ISC, see `TRADEMARKS.md`) tinted iOS system
    /// green — evocative of movement without colliding with Apple
    /// Health's red.
    private static let svg: String = """
    <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="#30D158" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><path d="M4 16v-2.38C4 11.5 2.97 10.5 3 8c.03-2.72 1.49-6 4.5-6C9.37 2 10 3.8 10 5.5c0 3.11-2 5.66-2 8.68V16a2 2 0 1 1-4 0Z"/><path d="M20 20v-2.38c0-2.12 1.03-3.12 1-5.62-.03-2.72-1.49-6-4.5-6C14.63 6 14 7.8 14 9.5c0 3.11 2 5.66 2 8.68V20a2 2 0 1 0 4 0Z"/><path d="M16 17h4"/><path d="M4 13h4"/></svg>
    """

    public static let dataUri: String =
        "data:image/svg+xml;base64," + Data(svg.utf8).base64EncodedString()
}

File: android/feature-activity-segments/src/main/kotlin/dev/omnesis/android/feature/activitysegments/ActivitySegmentsIcon.kt
// SPDX-License-Identifier: AGPL-3.0-or-later
// Copyright (c) 2026 Adrien Conrath

package dev.omnesis.android.feature.activitysegments

import java.util.Base64

/**
 * The android-activity-segments source's glyph — two waypoints joined by a
 * route, tinted Android green and embedded as a base64 data URI so the
 * portal's strict CSP (`img-src 'self' data: blob:`) lets it through. Same
 * construction shape as `CallLogIcon`/`AppUsageIcon`.
 */

private const val ROUTE_SVG =
    """<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><circle cx="6" cy="18" r="2"/><circle cx="18" cy="6" r="2"/><path d="M7.5 16.5 16.5 7.5"/></svg>"""

/** Android green. */
private const val ACTIVITY_SEGMENTS_TINT = "#3DDC84"

internal fun tintSvg(svg: String, color: String): String =
    svg.replace("stroke=\"currentColor\"", "stroke=\"$color\"")

internal fun svgDataUri(svg: String): String =
    "data:image/svg+xml;base64," + Base64.getEncoder().encodeToString(svg.toByteArray(Charsets.UTF_8))

/** Rides every sync-state write as the source's display icon. */
val ACTIVITY_SEGMENTS_ICON_DATA_URI: String = svgDataUri(tintSvg(ROUTE_SVG, ACTIVITY_SEGMENTS_TINT))
